ntoskrnl/mm/section.c

```diff
@@ -3770,7 +3770,8 @@ MiRosUnmapViewOfSection(
     }
 
     /* Notify debugger */
-    if (ImageBaseAddress && !SkipDebuggerNotify) DbgkUnMapViewOfSection(ImageBaseAddress);
+    if (ImageBaseAddress && !SkipDebuggerNotify && ImageSectionObject->MapCount == 0)
```

Comment:
`ImageSectionObject` may not have been initialized...

Reply:
Should be fixed now. Thanks.

Reply:
Instead of checking both `ImageBaseAddress and the `MapCount` here,
you should use the return value of `InterlockedDecrement(&ImageSectionObject->MapCount);`

One way of doing this is restoring the original code,
but using the result of the `InterlockedDecrement` and the `!SkipDebuggerNotify` to conditionally set `ImageBaseAddress` to `NULL`.